contracts/cw-ics721-bridge/src/ibc_helpers.rs

```diff
@@ -0,0 +1,185 @@
+use crate::{
+    ibc::{NonFungibleTokenPacketData, IBC_VERSION},
+    ContractError,
+};
+
+use cosmwasm_std::{
+    from_binary, to_binary, Binary, IbcAcknowledgement, IbcChannel, IbcEndpoint, IbcOrder,
+    StdError, StdResult,
+};
+
+/// Tries to remove the source prefix from a given class_id. If the
+/// class_id does not begin with the given prefix, returns
+/// `None`. Otherwise, returns `Some(unprefixed)`.
+pub(crate) fn try_pop_source_prefix<'a>(
+    source: &IbcEndpoint,
+    class_id: &'a str,
+) -> Option<&'a str> {
+    let source_prefix = get_endpoint_prefix(source);
+    // This must not panic in the face of non-ascii, or empty
+    // strings. We can not trust classID as it comes from an external
+    // IBC connection.
+    class_id.strip_prefix(&source_prefix)
+}
+
+/// Gets the classID prefix for a given IBC endpoint.
+pub(crate) fn get_endpoint_prefix(source: &IbcEndpoint) -> String {
+    format!("{}/{}/", source.port_id, source.channel_id)
+}
+
+/// Success ACK. 0x01 base64 encoded. By 0x01 base64 encoded, this
+/// literally means it is the base64 encoding of the number 1. You can
+/// test this by pasting this into a base64 decoder and, if it's for
+/// text, it'll output ascii character "START OF HEADING".
+pub fn ack_success() -> Binary {
+    // From the spec:
+    //
+    // > "Note that ... NonFungibleTokenPacketAcknowledgement must be
+    // > JSON-encoded (not Protobuf encoded) when serialized into packet
+    // > data."
+    //
+    // As such we encode '"AQ=="' as in JSON strings are surrounded by
+    // quotation marks as 'AQ==' is the base64 encoding of the number
+    // 1. The binary (ASCII code point list) version of this is below
+    // as we are dealing with a constant value.
+    Binary::from([34, 65, 81, 61, 61, 34])
```

Comment:
Lol, glad you wrote a test for this.

Reply:
I just couldn't bear unwrapping `to_binary` in here if we didn't have to.